### ISSUE TITLE
[PropertyInfo] Not compatible with phpDocumentor v5

### ISSUE THREAD

**Issue by martinatseequent** [symfony/symfony#36049]
**Symfony version(s) affected**: v5.0.5 (all?)

**Description**  
Using the property-info component in combination with phpDocumentor v5 (phpdocumentor/reflection-docblock  5.1.0) results in exceptions being thrown when unknown/invald tags are parsed. Looks like in the latest version og phpDocumentor these tags are not skipped, but an instance of `InvalidTag` is returned.

```
Error: Call to undefined method phpDocumentor\Reflection\DocBlock\Tags\InvalidTag::getType()

.../vendor/symfony/property-info/Extractor/PhpDocExtractor.php:145
```

**How to reproduce**  
```
<?php

include 'vendor/autoload.php';

use Symfony\Component\PropertyInfo\Extractor\PhpDocExtractor;

class SomeClass {
    /** @var  */
    public $foo;
}

$phpDocExtractor = new PhpDocExtractor();
$phpDocExtractor->getTypes(\SomeClass::class, 'foo');
```
**Possible Solution**  
Check for the value type of all values returned by `$docBlock->getTagsByName($tag)`

**Comment by nicolas-grekas** [comment]
Could you work on a patch to fix this please?

**Comment by JHoelzel** [comment]
So the Answer is to replace the line in phpdoc extractor
from
`             if ($tag!= null && null !== $tag->getType()) `
 with 
`             if ($tag!= null && !is_a($tag, 'phpDocumentor\Reflection\DocBlock\Tags\InvalidTag') &&null !== $tag->getType()) `

well actually it would be nice to know the tag but this temp fixes it ;)

**Comment by JHoelzel** [comment]
So i added this:
`if(is_a($tag, 'phpDocumentor\Reflection\DocBlock\Tags\InvalidTag')){
                throw new \InvalidArgumentException(sprintf('The Tag: "%s"is not a valid tag.', $tag));
            }`
before the code in question and the source of my problem is an array of an obkject.
To be exact:
 * @var []Person

Whereas 
* @var Person
of course does not reproduce the error

**Comment by JHoelzel** [comment]
i also created a bug report in the correct repo:
https://github.com/phpDocumentor/phpDocumentor/issues/2317

**Comment by DerManoMann** [comment]
Maybe this is the intended behaviour? Reporting an invalid tag rather than swallowing it silently?

**Comment by jaapio** [comment]
Hi @JHoelzel,

The usage of the @var you are defining is invalid according to the standards we wrote and are implemented in all tools using these standards.

From a symfony point of view, just read the changelogs. v5 is not fully compatible with v4. This is by purpose, because previously invalid tags where just ignored. There was no way to report them. So the implementation of symfony should be improved.

@nicolas-grekas please ping me at the symfony slack if you have any questions. I am available for help. Maybe I can come up with a patch for symfony but I can't say when that will be.

**Comment by JHoelzel** [comment]
Thank you i have just overtaken an older codebase and will simply be removing the comments in question.

although i do think a more explicit error would be nice ;)


thank you for your time

**Comment by ninsky** [comment]
Regarding the first post:
>Symfony version(s) affected: v5.0.5 **(all?)**

This also affects **Symfony 4.4.7**. What makes me wonder is, that I've used Symfony 4.4.6 before which didn't have this issue.  

See [this stacktrace](https://gist.github.com/ninsky/17bda9c9b6e65b34c47a5a995a7742b2). Hope this helps.

**Comment by vlad57** [comment]
Hi,

I have the the same issue. I just updated packages with composer update and got this error when I'm trying to access a page with Form. 

<img width="1673" alt="Capture d’écran 2020-04-06 à 11 31 07" src="https://user-images.githubusercontent.com/18573676/78551796-9d27f580-7806-11ea-83d4-c9cc07a7e103.png">

as @JHoelzel said, replacing

`if ($tag!= null && null !== $tag->getType())`
by
`if ($tag!= null && !is_a($tag, 'phpDocumentor\Reflection\DocBlock\Tags\InvalidTag') &&null !== $tag->getType())` works well temporary.

**Comment by rp3e11** [comment]
Hi,

I have a similar problem that occurred after the last update. In my case though the code fails at a getDescription on an InvalidTag. 
```php
        foreach ($docBlock->getTagsByName('var') as $var) {

            $varDescription = $var->getDescription()->render();

            if (!empty($varDescription)) {
                return $varDescription;
            }
        }
```
If the var tag is not correctly defined it returns `$var` is of type InvalidTag for which the function getDescription does not exist. It is clear the @var declaration in my code has to be fixed but it would be helpful to know which class and property are affected. I am new to symfony so I am not sure about possible side effects and what expection classes you would use, but something along those lines would have helped me to debug my code 
```php
        foreach ($docBlock->getTagsByName('var') as $var) {
            if (is_a($var, 'phpDocumentor\Reflection\DocBlock\Tags\InvalidTag')) {
                throw new DefinitionException('Failed to get the description of the @var tag "' . $property . '" for class "' . $class . '". Please check that the @var tag is correctly defined');
            }
            $varDescription = $var->getDescription()->render();

            if (!empty($varDescription)) {
                return $varDescription;
            }
        }
```

![image](https://user-images.githubusercontent.com/13377340/79062997-3475c880-7c9f-11ea-99d6-09c4f8405044.png)

**Comment by bianca-ioana** [comment]
> **Symfony version(s) affected**: v5.0.5 (all?)
> 
> **Description**
> Using the property-info component in combination with phpDocumentor v5 (phpdocumentor/reflection-docblock 5.1.0) results in exceptions being thrown when unknown/invald tags are parsed. Looks like in the latest version og phpDocumentor these tags are not skipped, but an instance of `InvalidTag` is returned.
> 
> ```
> Error: Call to undefined method phpDocumentor\Reflection\DocBlock\Tags\InvalidTag::getType()
> 
> .../vendor/symfony/property-info/Extractor/PhpDocExtractor.php:145
> ```
> 
> **How to reproduce**
> 
> ```
> <?php
> 
> include 'vendor/autoload.php';
> 
> use Symfony\Component\PropertyInfo\Extractor\PhpDocExtractor;
> 
> class SomeClass {
>     /** @var  */
>     public $foo;
> }
> 
> $phpDocExtractor = new PhpDocExtractor();
> $phpDocExtractor->getTypes(\SomeClass::class, 'foo');
> ```
> 
> **Possible Solution**
> Check for the value type of all values returned by `$docBlock->getTagsByName($tag)`
It also affects sf version 4.4

**Comment by jaapio** [comment]
Actually this effects all symfony versions if v5 is allowed by the composer.json. But I didn't check which versions do have a requirement on v5. There is a reason why we tagged it v5. Because of this BC break.

**Comment by spmsupun** [comment]
Any news on this ?

**Comment by BTSmolders** [comment]
Could this fix also be applied to a 4.4 release. I can see that the 4.4 branch contains the fix, but the latest release (4.4.10) does not contain the fix.
